qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+                          'state': interlock_status[int(float(state))]}
+            states.append(state_dict)
+        return tuple(states)
+
+    def get_ip_address(self) -> str:
+        """
+        Returns the current IP address of the instrument.
+        """
+        return self.ask('lan.status.ipaddress')
+
+    def reset_local_network(self) -> None:
+        """
+        Resets the local network (LAN).
+        """
+        self.write('lan.reset()')
+
+    def save_setup(self, val: Optional[str] = None) -> None:
+        """
+        Saves the present setup.
+
+        Args:
+            val: An optional string representing the path and the file name
+                to which the setup shall be saved on a USB flash drive. If not
+                provided, the setup will be saved to the nonvolatile memory
+                of the instrument. Any previous saves shall be overwritten.
+        """
+        if val is not None:
+            self.write(f"setup.save('{val}')")
+        else:
+            self.write(f'setup.save()')
+
+    def load_setup(self, val: Union[int, str]) -> None:
+        """
+        Loads the settings from a saved setup.
+
+        Args:
+            val: An integer or a string that specifies the location of saved
+                setup. If it is `0`, factory defaults load. If it is `1`,
+                the saved setup from the nonvolatile memory is recalled.
+                Otherwise, a string specifying the relative path to the saved
+                setup on a USB drive should be passed in.
+        """
+        self.write(f"setup.recall('{val}')")
+
+    def _validator(self, val: str) -> bool:
+        """
+        Instrument specific validator. As the number of validation points
+        are around 15k, to avoid QCoDeS parameter validation to print them all,
+        we shall raise a custom exception.
+        """
+        ch = self.get_channels()
+        ch_range = self._get_channel_ranges()
+        slots = ['allslots', *self._get_slot_name()]
+        backplanes = self.get_analog_backplane_specifiers()
+        specifier = val.split(',')
+        for element in specifier:
+            if element not in (*ch, *ch_range, *slots, *backplanes):
+                return False
+        return True
+
+    def connect_message(self, idn_param: str = 'IDN',
+                        begin_time: float = None) -> None:
+        """
+        Overwrites the generic QCoDeS instrument connect message.
+        Here, additionally, we provide information about
+        which slots of the system switch is occupied with what
+        kind of matrix, as well.
+        """
+        idn = self.get_idn()
+        cards = self.get_switch_cards()
+        states = self.get_interlock_state()
+
+        con_msg = ('Connected to: {vendor} {model} SYSTEM SWITCH '
+                   '(serial:{serial}, firmware:{firmware})'.format(**idn))
+        print(con_msg)
+        self.log.info(f"Connected to instrument: {idn}")
```

Comment:
these three lines can be substituted with `super().connect_message()`, right?

Reply:
No, because "SYSTEM SWITCH" is a custom addition.

Reply:
hmm, do we care about this custom addition? :)